depscan/cli.py

```diff
@@ -432,10 +432,43 @@ def export_bom(bom_data, pkg_vulnerabilities, vdr_file):
     # Update the tools section
     if isinstance(tools, dict):
         bom_data = summarise_tools(tools, metadata, bom_data)
+    if bom_data.get("components"):
+        bom_data = remove_extra_properties(bom_data)
     bom_data["vulnerabilities"] = pkg_vulnerabilities
     json_dump(vdr_file, bom_data, error_msg=f"Unable to generate VDR file at {vdr_file}", log=LOG)
 
 
+def remove_extra_properties(bom_data):
```

Comment:
@timmyteo Could we refactor this to simplify?

```python
exclude_properties = {"Namespaces", "ImportedModules"}
for i, component in enumerate(bom_data["components"]):
    if properties := component.get("properties"):
         bom_data["components"][i]["properties"] = [p for p in properties if p.get("name") not in exclude_properties]
return bom_data
``` 

@prabhu Do we want to make the excluded or allowed properties an optional argument for this function?

Reply:
Optional argument sounds good. Support for regex or startswith since blint uses "internal:" prefix